components/panel/body.js

```diff
@@ -52,7 +52,10 @@ class PanelBody extends Component {
 							aria-expanded={ isOpened }
 						>
 							<Dashicon icon={ icon } />
-							{ title }
+							<span className="components-panel__body-title__text-span">
+								{ title }
+							</span>
+							{customTitleArea}
```

Comment:
Instead of an extra `customTitleArea` prop, could the wrapping `PanelColor` component render the `title` with both the text span and color span?

```jsx
<PanelBody
	{ ...props }
	title={ [
		<span key="title">{ title }</span>,
		<span key="color" style={ { background: colorValue } }>,
	] }
/>
```